### PR TITLE
kam describe

### PR DESCRIPTION
**Pull request by JainamShah-08**
**What type of PR is this?**
> /kind enhancement

**What does this PR do / why we need it**:
The PR provides `kam describe command`
**Have you updated the necessary documentation?**

* [✅ ] Documentation has been updated.

**Which issue(s) this PR fixes**:
GITOPS-2151

**How to test changes / Special notes to the reviewer**:

- `kam describe --help` - to see the flags available for the command.
- `kam describe --application-folder <Path to application>` - command should display list of components in application and environments within the component.

### CODE REVIEW

File: pkg/cmd/component/describe.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	}
+	exists, _ := ioutils.IsExisting(appFS, io.Output)
+	if !exists {
+		return fmt.Errorf("the given Path : %s  doesn't exists ", io.Output)
+	}
+	exists, _ = ioutils.IsExisting(appFS, filepath.Join(io.Output, io.ApplicationName))
+	if !exists {
+		return fmt.Errorf("the given Application: %s  doesn't exists in the Path: %s", io.ApplicationName, io.Output)
+	}
+	exists, _ = ioutils.IsExisting(appFS, filepath.Join(io.Output, io.ApplicationName, io.ComponentName))
```
Comment (keithchong) [inline pkg/cmd/component/describe.go]: Do we need this?

File: pkg/cmd/component/describe.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+		if flags[flag] == "" {
+			missingFlags = append(missingFlags, fmt.Sprintf("%q", flag))
+		}
+	}
+	if len(missingFlags) > 0 {
+		return missingFlagErr(missingFlags)
+	}
+	return nil
+}
+func checkENv(path string) []string {
```
Comment (keithchong) [inline pkg/cmd/component/describe.go]: Spelling

File: pkg/cmd/component/describe.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+		Use:     name,
+		Short:   describeShortDescC,
+		Long:    describeLongDescC,
+		Example: fmt.Sprintf(describeExampleC, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			genericclioptions.GenericRun(o, cmd, args)
+		},
+	}
+	descibeCmd.Flags().StringVar(&o.Output, "output", "./", "Path to write GitOps resources")
+	descibeCmd.Flags().StringVar(&o.ApplicationName, "application-name", "", "Provide a name of application")
```
Comment (keithchong) [inline pkg/cmd/component/describe.go]: How about, "Provide the name of the application"

The application supposedly has already been created

File: pkg/cmd/component/describe.go (hunk 2 above)
Comment (JainamShah-08) [reply pkg/cmd/component/describe.go]: Made the change

File: pkg/cmd/component/describe.go (hunk 1 above)
Comment (JainamShah-08) [reply pkg/cmd/component/describe.go]: No. I have deleted it in the new commit

File: pkg/cmd/component/describe.go (hunk 3 above)
Comment (JainamShah-08) [reply pkg/cmd/component/describe.go]: Changed

**Review by keithchong — APPROVED** [review]
LGTM

**Pull request merged**